### PR TITLE
Moved utils to programs

### LINKED ISSUE #72404: Embed utils into the main clickhouse binary
Several of the tools put under `utils` are useful, written in C++ and with the CH code as base, such as `keeper-data-dumper`.

We should put those tools under programs instead and distribute them as part of the main clickhouse binary.

### PR DESCRIPTION
**Pull request by nikitamikhaylov**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
All small utilities previously stored in `/utils` folder and required manual compilation from sources are now a part of main ClickHouse bundle. This closes: https://github.com/ClickHouse/ClickHouse/issues/72404

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Run only fuzzers related jobs (libFuzzer fuzzers, AST fuzzers, etc.)
- [ ]  Exclude: AST fuzzers
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

File: utils/zookeeper-cli/zookeeper-cli.cpp
Comment (nikitamikhaylov) [inline utils/zookeeper-cli/zookeeper-cli.cpp:1]: I removed this, because we have a shiny `keeper-client`.

**Comment by robot-ch-test-poll1** [comment]
*This is an automated comment for commit 20b572f4a8b957a23be015b922468a9354a0776a with description of existing statuses. It's updated for the latest CI running*

[❌ Click here](https://s3.amazonaws.com/clickhouse-test-reports/72426/20b572f4a8b957a23be015b922468a9354a0776a/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Style check</td><td>Runs a set of checks to keep the code style clean. If some of tests failed, see the related log from the report</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/72426/20b572f4a8b957a23be015b922468a9354a0776a/style_check.html">❌ failure</a></td></tr>
<tbody>
</table>

File: base/harmful/harmful.c [hunk 1]
```diff
@@ -65,7 +65,6 @@ TRAP(gethostbyaddr)
 TRAP(gethostbyname)
 TRAP(gethostbyname2)
 TRAP(gethostent)
-TRAP(getlogin)
```
Comment (Algunenano) [inline base/harmful/harmful.c:68]: Where do we use it? It's not thread safe and can cause races with other functions:

```
ATTRIBUTES         [top](https://man7.org/linux/man-pages/man3/getlogin.3.html#top_of_page)

       For an explanation of the terms used in this section, see
       [attributes(7)](https://man7.org/linux/man-pages/man7/attributes.7.html).
       ┌──────────────┬───────────────┬────────────────────────────────┐
       │ Interface    │ Attribute     │ Value                          │
       ├──────────────┼───────────────┼────────────────────────────────┤
       │ getlogin()   │ Thread safety │ MT-Unsafe race:getlogin        │
       │              │               │ race:utent sig:ALRM timer      │
       │              │               │ locale                         │
       ├──────────────┼───────────────┼────────────────────────────────┤
       │ getlogin_r() │ Thread safety │ MT-Unsafe race:utent sig:ALRM  │
       │              │               │ timer locale                   │
       ├──────────────┼───────────────┼────────────────────────────────┤
       │ cuserid()    │ Thread safety │ MT-Unsafe race:cuserid/!string │
       │              │               │ locale                         │
       └──────────────┴───────────────┴────────────────────────────────┘

       In the above table, utent in race:utent signifies that if any of
       the functions [setutent(3)](https://man7.org/linux/man-pages/man3/setutent.3.html), [getutent(3)](https://man7.org/linux/man-pages/man3/getutent.3.html), or [endutent(3)](https://man7.org/linux/man-pages/man3/endutent.3.html) are used
       in parallel in different threads of a program, then data races
       could occur.  getlogin() and getlogin_r() call those functions,
       so we use race:utent to remind users.
```

**Review by Algunenano — COMMENTED** [review]
Had some question about removing functions from the harmful list. I'd rather not do it if possible

**Review by Algunenano — APPROVED** [review]

**Pull request merged**